mpf/core/config_processor.py

```diff
@@ -179,7 +179,7 @@ def _load_config_file_and_return_loaded_files(
 
         self.log.info('Loading config: %s', filename)
 
-        if config_type in ("machine", "mode"):
+        if config_type in ("machine", "mode") and config_spec:
```

Comment:
This is for my own personal use. I have an automation script that imports and parses the config files, but a recent refactor of _config_processor_ changed the behavior and expects that the _config_spec_ has been loaded into memory. My automation script doesn't do this, so I need to ignore config spec parsing in that case.

Reply:
I see that. I recently added a config reference parser to expose a stable interface for config reference. It is already used in mpf language server and mpf docs. Guess that might make sense for you too.

Out of curiosity: What does your script do?

Reply:
A config parser is exactly what I'm after! Could you point to a reference of it in use so I know where to start?

My script is for audio management, since I have a very large library of sfx to pull from and I constantly add sounds, remove sounds, and move sounds from one mode to another. My script traverses the file trees (once for my source library, once for my machine folder) and traverses the config files, then moves all the sound assets into the appropriate folders.

Reply:
It will also do a bulk export of all audio files for the machine, which is helpful since I don't track those in git. When I setup MPF on a different machine, this script's import/export comes in real handy :)

https://github.com/avanwinkle/masseffect2/blob/release/0.52/mesound.py

Reply:
Sounds interesting. Might be interesting for other users as well. We could turn this into MPF commands if you want.

Just checked the code. I only moved the event parser so far. Here is the config parsing from the mpf language server: https://github.com/missionpinball/mpf-ls/blob/dev/mpfls/workspace.py